crates/ide_assists/src/handlers/inline_type_alias.rs

```diff
@@ -0,0 +1,747 @@
+// Some ideas for future improvements:
+// - Support replacing aliases which are used in expressions, e.g. `A::new()`.
+// - "inline_alias_to_users" assist #10881.
+// - Remove unused aliases if there are no longer any users, see inline_call.rs.
+
+use hir::PathResolution;
+use itertools::Itertools;
+use std::collections::HashMap;
+use syntax::{
+    ast::{
+        self,
+        make::{self},
+        HasGenericParams, HasName,
+    },
+    ted::{self},
+    AstNode, NodeOrToken, SyntaxNode,
+};
+
+use crate::{
+    assist_context::{AssistContext, Assists},
+    AssistId, AssistKind,
+};
+
+// Assist: inline_type_alias
+//
+// Replace a type alias with its concrete type.
+//
+// ```
+// type A<T = u32> = Vec<T>;
+//
+// fn main() {
+//     let a: $0A;
+// }
+// ```
+// ->
+// ```
+// type A<T = u32> = Vec<T>;
+//
+// fn main() {
+//     let a: Vec<u32>;
+// }
+// ```
+pub(crate) fn inline_type_alias(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let alias_instance = ctx.find_node_at_offset::<ast::PathType>()?;
+    let alias = get_type_alias(&ctx, &alias_instance)?;
+    let concrete_type = alias.ty()?;
+
+    let replacement = if let Some(alias_generics) = alias.generic_param_list() {
+        if alias_generics.generic_params().next().is_none() {
+            cov_mark::hit!(no_generics_params);
+            return None;
+        }
+
+        let instance_args =
+            alias_instance.syntax().descendants().find_map(ast::GenericArgList::cast);
+
+        create_replacement(
+            &LifetimeMap::new(&instance_args, &alias_generics)?,
+            &ConstAndTypeMap::new(&instance_args, &alias_generics)?,
+            &concrete_type,
+        )
+    } else {
+        concrete_type.to_string()
+    };
```

Comment:
Let's move this into the closure, except for the early return.

Reply:
There are two more early exits for `LifetimeMap::new` and `ConstAndTypeMap::new`. Would you prefer this?
```rust

    enum Replacement {
        Generic { lifetime_map: LifetimeMap, const_and_type_map: ConstAndTypeMap },
        Plain,
    }

    let replacement = if let Some(alias_generics) = alias.generic_param_list() {
        if alias_generics.generic_params().next().is_none() {
            cov_mark::hit!(no_generics_params);
            return None;
        }

        let instance_args =
            alias_instance.syntax().descendants().find_map(ast::GenericArgList::cast);

        Replacement::Generic {
            lifetime_map: LifetimeMap::new(&instance_args, &alias_generics)?,
            const_and_type_map: ConstAndTypeMap::new(&instance_args, &alias_generics)?,
        }
    } else {
        Replacement::Plain
    };

    let target = alias_instance.syntax().text_range();

    acc.add(
        AssistId("inline_type_alias", AssistKind::RefactorInline),
        "Inline type alias",
        target,
        |builder| {
            let replacement_text = match replacement {
                Replacement::Generic { lifetime_map, const_and_type_map } => {
                    create_replacement(&lifetime_map, &const_and_type_map, &concrete_type)
                }
                Replacement::Plain => concrete_type.to_string(),
            };

            builder.replace(target, replacement_text);
        },
    )
```

Reply:
Ye that would work